helm/cloud-provider-azure/templates/cloud-provider-azure.yaml

```diff
@@ -443,7 +443,16 @@ data:
 
     ((Get-Content -path $env:CONTAINER_SANDBOX_MOUNT_POINT/var/lib/cpaw/kubeconfig.conf -Raw) -replace '/var',"$($env:CONTAINER_SANDBOX_MOUNT_POINT)/var") | Set-Content -Path $env:CONTAINER_SANDBOX_MOUNT_POINT/var/lib/cpaw/kubeconfig.conf
 
-    Invoke-Expression "$cm --node-name=$env:NODE_NAME --use-instance-metadata=true --kubeconfig=$env:CONTAINER_SANDBOX_MOUNT_POINT/var/lib/cpaw/kubeconfig.conf"
+    $argList = @()
+
+    foreach ($var in @("CLOUD_CONFIG", "KUBE_API_BURST", "KUBE_API_CONTENT_TYPE", "KUBE_API_QPS", "MASTER", "MIN_RESYNC_PERIOD", "NODE_STATUS_UPDATE_FREQUENCY", "WAIT_ROUTES")) {
+      if ([System.Environment]::GetEnvironmentVariable($var) -NE $null) {
+        $argList += [System.Environment]::GetEnvironmentVariable($var)
+      }
+    }
+    
+    Write-Host "Running $cm $argList --node-name=$env:NODE_NAME --use-instance-metadata=true --kubeconfig=$env:CONTAINER_SANDBOX_MOUNT_POINT/var/lib/cpaw/kubeconfig.conf"
```

Comment:
@marosset do you think it's helpful to keep this for later debugging or should I get rid of it?

Reply:
I always find these messages helpful so my vote is to keep it